src/web_interface/components/ajax_routes.py

```diff
@@ -27,16 +27,21 @@ def _init_component(self):
 
     @roles_accepted(*PRIVILEGES['view_analysis'])
     def _ajax_get_tree_children(self, uid, root_uid=None, compare_id=None):
-        with ConnectTo(CompareDbInterface, self._config) as sc:
-            exclusive_files = sc.get_exclusive_files(compare_id, root_uid)
+        exclusive_files = self._get_exclusive_files(compare_id, root_uid)
         tree = self._generate_exclusive_files_tree(exclusive_files, root_uid, uid)
         children = [
             self._generate_jstree_node(child_node)
             for child_node in tree.get_list_of_child_nodes()
         ]
         return jsonify(children)
 
-    def _generate_exclusive_files_tree(self, exclusive_files, root_uid, uid):
+    def _get_exclusive_files(self, compare_id, root_uid):
+        if compare_id:
+            with ConnectTo(CompareDbInterface, self._config) as sc:
+                return sc.get_exclusive_files(compare_id, root_uid)
+        return None
+
+    def _generate_exclusive_files_tree(self, exclusive_files, root_uid, uid) -> FileTreeNode:
```

Comment:
the wording is misleading, since it is not just used for the exclusive file tree.

` def _generate_files_tree(self, whitelist, root_uid, uid) -> FileTreeNode:`

we should add a docstring explaining "the whitlist" feature.